internal/otelcollector/config/metricagent/config_builder.go

```diff
@@ -283,6 +283,9 @@ func (b *Builder) addFilterDropVirtualNetworkInterfacesProcessor() buildComponen
 	)
 }
 
+// TODO(TeodorSAP): The Prometheus receiver sets the service.name attribute by default.
+// We currently remove it here, but we should investigate configuring the receiver
+// to not set this attribute in the first place for better efficiency.
```

Comment:
It has nothing to do with efficiency. The Prometheus receiver sets service.name to the scrape job name, which prevents it from being enriched by service name processor.